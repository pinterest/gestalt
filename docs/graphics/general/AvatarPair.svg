<svg xmlns="http://www.w3.org/2000/svg" width="312" height="180" fill="none"><circle cx="139" cy="84" r="35" fill="#005C62"/><path fill="#CDF9CD" d="M138.789 86.144v-2.895h-9.458v3.48h4.885l-.015.427c-.094 2.848-2.232 4.762-5.323 4.762-3.714 0-6.055-3.022-6.055-7.99 0-4.872 2.232-7.862 5.868-7.862 2.622 0 4.511 1.36 5.213 3.75h4.682c-.655-4.731-4.573-7.816-9.895-7.816C122.136 72 118 76.635 118 83.976 118 91.428 122.105 96 128.8 96c6.15 0 9.989-3.797 9.989-9.856Zm14.453 9.271v-18.89H160v-3.94h-18.23v3.94h6.758v18.89h4.714Z"/><path fill="#fff" fill-rule="evenodd" d="M139 117c18.225 0 33-14.775 33-33s-14.775-33-33-33-33 14.775-33 33 14.775 33 33 33Zm0 2c19.33 0 35-15.67 35-35s-15.67-35-35-35-35 15.67-35 35 15.67 35 35 35Z" clip-rule="evenodd"/><circle cx="174" cy="96" r="35" fill="#005C62"/><path fill="#CDF9CD" d="M158 96c0-8.838 7.163-16 16-16s16 7.162 16 16c0 8.837-7.163 16-16 16s-16-7.163-16-16Zm10.256 4.344a1 1 0 0 0-1.412-.1.999.999 0 0 0-.1 1.412c1.387 1.6 3.812 3.344 7.2 3.344 3.5 0 5.919-1.744 7.312-3.344a.999.999 0 0 0-.1-1.412 1 1 0 0 0-1.412.1c-1.125 1.294-3.032 2.656-5.8 2.656-2.656 0-4.569-1.362-5.688-2.656Zm.769-9.344c-1.106 0-2 .894-2 2 0 1.106.894 2 2 2 1.1 0 2-.894 2-2 0-1.106-.9-2-2-2Zm10 4c1.1 0 2-.894 2-2 0-1.106-.9-2-2-2-1.106 0-2 .894-2 2 0 1.106.894 2 2 2Z"/><path fill="#fff" fill-rule="evenodd" d="M174 129c18.225 0 33-14.775 33-33s-14.775-33-33-33-33 14.775-33 33 14.775 33 33 33Zm0 2c19.33 0 35-15.67 35-35s-15.67-35-35-35-35 15.67-35 35 15.67 35 35 35Z" clip-rule="evenodd"/></svg>