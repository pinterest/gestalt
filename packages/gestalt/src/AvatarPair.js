// @flow strict
import * as React from 'react';
import PropTypes from 'prop-types';
import Avatar from './Avatar.js';
import Box from './Box.js';

type Props = {|
  collaborators: Array<{|
    name: string,
    src?: string,
  |}>,
  size?: 'md' | 'lg' | 'fit',
|};

const sizes = {
  md: 48,
  lg: 64,
};

export default function AvatarPair({ collaborators, size = 'fit' }: Props) {
  const width = size === 'fit' ? '100%' : sizes[size];
  return (
    <Box position="relative" width={width}>
      <Box dangerouslySetInlineStyle={{ __style: { paddingBottom: '100%' } }} />
      {(collaborators || []).slice(0, 2).map(({ name, src }, index) => (
        <Box
          key={`${name}-${index}`}
          position="absolute"
          height="75%"
          width="75%"
          dangerouslySetInlineStyle={{
            __style: {
              left: index === 0 ? 0 : '25%',
              top: index === 0 ? 0 : '25%',
            },
          }}
        >
          <Avatar src={src} name={name} outline />
        </Box>
      ))}
    </Box>
  );
}

AvatarPair.propTypes = {
  collaborators: PropTypes.arrayOf(
    PropTypes.exact({
      name: PropTypes.string.isRequired,
      src: PropTypes.string,
    })
  ).isRequired,
  size: PropTypes.oneOf(['md', 'lg', 'fit']),
};
