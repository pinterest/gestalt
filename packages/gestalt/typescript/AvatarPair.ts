import type { Node } from "react";
import type { CollaboratorDataType } from "./Avatar";
import Avatar from "./Avatar";
import Box from "./Box";
type Props = {
  collaborators: ReadonlyArray<CollaboratorDataType>;
  size?: "md" | "lg" | "fit";
};
const sizes = {
  md: 48,
  lg: 64,
};
/**
 * https://gestalt.pinterest.systems/AvatarPair
 */

export default function AvatarPair({
  collaborators,
  size = "fit",
}: Props): Node {
  const width = size === "fit" ? "100%" : sizes[size];
  return (
    <Box position="relative" width={width}>
      <Box
        dangerouslySetInlineStyle={{
          __style: {
            paddingBottom: "100%",
          },
        }}
      />
      {(collaborators || []).slice(0, 2).map(({ name, src }, index) => (
        <Box
          key={`${name}-${index}`}
          position="absolute"
          height="75%"
          width="75%"
          dangerouslySetInlineStyle={{
            __style: {
              left: index === 0 ? 0 : "25%",
              top: index === 0 ? 0 : "25%",
            },
          }}
        >
          <Avatar src={src} name={name} outline />
        </Box>
      ))}
    </Box>
  );
}