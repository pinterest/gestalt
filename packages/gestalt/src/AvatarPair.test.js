// @flow strict
import React from 'react';
import { create } from 'react-test-renderer';
import AvatarPair from './AvatarPair.js';

describe('AvatarPair', () => {
  it('Renders renders 2 collaborators', () => {
    const component = create(
      <AvatarPair
        collaborators={[
          {
            name: 'Keerthi',
            src: 'https://i.ibb.co/ZfCZrY8/keerthi.jpg',
          },
          {
            name: 'Shanice',
            src: 'https://i.ibb.co/7tGKGvb/shanice.jpg',
          },
        ]}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders renders the avatar with text when no image is provided', () => {
    const component = create(
      <AvatarPair
        collaborators={[
          {
            name: 'Keerthi',
            src: 'https://i.ibb.co/ZfCZrY8/keerthi.jpg',
          },
        ]}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Renders only the first 2 collobarators when multiple are passed in', () => {
    const component = create(
      <AvatarPair
        collaborators={[
          {
            name: 'Keerthi',
            src: 'https://i.ibb.co/ZfCZrY8/keerthi.jpg',
          },
          {
            name: 'Shanice',
            src: 'https://i.ibb.co/7tGKGvb/shanice.jpg',
          },
          {
            name: 'James',
            src: 'https://i.ibb.co/7tGKGvb/james.jpg',
          },
        ]}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
