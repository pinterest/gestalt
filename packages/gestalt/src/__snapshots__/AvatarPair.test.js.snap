// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarPair Renders only the first 2 collobarators when multiple are passed in 1`] = `
<div
  className="box relative"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="box"
    style={
      Object {
        "paddingBottom": "100%",
      }
    }
  />
  <div
    className="absolute box"
    style={
      Object {
        "height": "75%",
        "left": 0,
        "top": 0,
        "width": "75%",
      }
    }
  >
    <div
      className="box circle relative whiteBg"
      style={
        Object {
          "boxShadow": "0 0 0 1px #fff",
          "height": "",
          "width": "100%",
        }
      }
    >
      <div
        className="Mask circle willChangeTransform"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      >
        <div
          className="box relative"
          style={
            Object {
              "backgroundColor": "#efefef",
              "paddingBottom": "100%",
            }
          }
        >
          <img
            alt="Keerthi"
            className="img"
            importance="auto"
            loading="auto"
            onError={[Function]}
            onLoad={[Function]}
            src="https://i.ibb.co/ZfCZrY8/keerthi.jpg"
          />
        </div>
        <div
          className="wash"
        />
      </div>
    </div>
  </div>
  <div
    className="absolute box"
    style={
      Object {
        "height": "75%",
        "left": "25%",
        "top": "25%",
        "width": "75%",
      }
    }
  >
    <div
      className="box circle relative whiteBg"
      style={
        Object {
          "boxShadow": "0 0 0 1px #fff",
          "height": "",
          "width": "100%",
        }
      }
    >
      <div
        className="Mask circle willChangeTransform"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      >
        <div
          className="box relative"
          style={
            Object {
              "backgroundColor": "#efefef",
              "paddingBottom": "100%",
            }
          }
        >
          <img
            alt="Shanice"
            className="img"
            importance="auto"
            loading="auto"
            onError={[Function]}
            onLoad={[Function]}
            src="https://i.ibb.co/7tGKGvb/shanice.jpg"
          />
        </div>
        <div
          className="wash"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AvatarPair Renders renders 2 collaborators 1`] = `
<div
  className="box relative"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="box"
    style={
      Object {
        "paddingBottom": "100%",
      }
    }
  />
  <div
    className="absolute box"
    style={
      Object {
        "height": "75%",
        "left": 0,
        "top": 0,
        "width": "75%",
      }
    }
  >
    <div
      className="box circle relative whiteBg"
      style={
        Object {
          "boxShadow": "0 0 0 1px #fff",
          "height": "",
          "width": "100%",
        }
      }
    >
      <div
        className="Mask circle willChangeTransform"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      >
        <div
          className="box relative"
          style={
            Object {
              "backgroundColor": "#efefef",
              "paddingBottom": "100%",
            }
          }
        >
          <img
            alt="Keerthi"
            className="img"
            importance="auto"
            loading="auto"
            onError={[Function]}
            onLoad={[Function]}
            src="https://i.ibb.co/ZfCZrY8/keerthi.jpg"
          />
        </div>
        <div
          className="wash"
        />
      </div>
    </div>
  </div>
  <div
    className="absolute box"
    style={
      Object {
        "height": "75%",
        "left": "25%",
        "top": "25%",
        "width": "75%",
      }
    }
  >
    <div
      className="box circle relative whiteBg"
      style={
        Object {
          "boxShadow": "0 0 0 1px #fff",
          "height": "",
          "width": "100%",
        }
      }
    >
      <div
        className="Mask circle willChangeTransform"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      >
        <div
          className="box relative"
          style={
            Object {
              "backgroundColor": "#efefef",
              "paddingBottom": "100%",
            }
          }
        >
          <img
            alt="Shanice"
            className="img"
            importance="auto"
            loading="auto"
            onError={[Function]}
            onLoad={[Function]}
            src="https://i.ibb.co/7tGKGvb/shanice.jpg"
          />
        </div>
        <div
          className="wash"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AvatarPair Renders renders the avatar with text when no image is provided 1`] = `
<div
  className="box relative"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="box"
    style={
      Object {
        "paddingBottom": "100%",
      }
    }
  />
  <div
    className="absolute box"
    style={
      Object {
        "height": "75%",
        "left": 0,
        "top": 0,
        "width": "75%",
      }
    }
  >
    <div
      className="box circle relative whiteBg"
      style={
        Object {
          "boxShadow": "0 0 0 1px #fff",
          "height": "",
          "width": "100%",
        }
      }
    >
      <div
        className="Mask circle willChangeTransform"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      >
        <div
          className="box relative"
          style={
            Object {
              "backgroundColor": "#efefef",
              "paddingBottom": "100%",
            }
          }
        >
          <img
            alt="Keerthi"
            className="img"
            importance="auto"
            loading="auto"
            onError={[Function]}
            onLoad={[Function]}
            src="https://i.ibb.co/ZfCZrY8/keerthi.jpg"
          />
        </div>
        <div
          className="wash"
        />
      </div>
    </div>
  </div>
</div>
`;
